src/codec/RowWriterV2.cpp

```diff
@@ -780,21 +780,28 @@ WriteResult RowWriterV2::write(ssize_t index, const Time& v) noexcept {
 WriteResult RowWriterV2::write(ssize_t index, const DateTime& v) noexcept {
     auto field = schema_->field(index);
     auto offset = headerLen_ + numNullBytes_ + field->offset();
+    int16_t year = v.year;
```

Comment:
Maybe we could store the bit field directly in next incompatible version.

Reply:
It might have trouble with the secondary index, but we could have a try.